### DISCUSSION TITLE
How do I retrieve spacy's compound words?

### DISCUSSION THREAD

**Discussion by AnnemarieWittig** [explosion/spaCy#11969] [Help: Coding & Implementations] [answered]
Hi!

I am currently trying to retrieve any compound words from my sentence. For example, I will have a sentence such as "This is Angela Merkel" and would get pos tags (simply printing the texts) such as: 

```console
This this PRON DT nsubj Xxxx True True
is be AUX VBZ ROOT xx True True
Angela Angela PROPN NNP compound Xxxxx True False
Merkel Merkel PROPN NNP attr Xxxxx True False
. . PUNCT . punct . False False
```

Now I know and can see using the visualizer that Angela is compound to Merkel, but I don't quite understand yet how I find this information using just Python. Could someone point me there? 

My test script: 
```python
import spacy
from spacy import displacy

nlp = spacy.load("en_core_web_sm")
doc = nlp("This is Angela Merkel.")

for token in doc:
    print(token.text, token.lemma_, token.pos_, token.tag_, token.dep_,
            token.shape_, token.is_alpha, token.is_stop)
```

answered in subthread / subcomment https://github.com/explosion/spaCy/discussions/11969#discussioncomment-4454080

**Comment by polm** [comment, marked as answer]
The `compound` relation is a dependency relation and is set by the (dependency) parser, you can access it with `token.dep_`. 

There's not a special function for getting words marked with the `compound` relation, but [`merge_noun_chunks`](https://spacy.io/api/pipeline-functions#merge_noun_chunks) should merge compound words (in addition to other chunks).

**Reply by AnnemarieWittig** [reply]
Thank you for your answer! However, it is not quite what I need.

Yes, accessing `token.dep_` is how I know that there are words marked as compounds (as seen in my output listed above) and from the visualization I know that the  compound "Merkel" points to "Angela":
![image](https://user-images.githubusercontent.com/59013332/208508061-0cc65c62-efbc-423b-83a0-d2a405873a51.png)
What I need to find out is how I can access that information in code. 
Something like (pseudo code):
```python
if STRING is compound:
   return all STRING2 that STRING is compound to
```

Obviously I might have more complex sentences in the actual situation, possibly containing multiple compound words which are not following right behind each other. Merging the words wouldn't really be working for me as I don't want or need them merged and still prefer them all separate.

**Reply by polm** [reply]
Ah, OK, thanks for explaining - that shouldn't be very hard then. You can do something like this:

```
for tok in doc:
    for child in tok.children:
        if child.dep_ == "compound":
            print(f"{child} is compound to {tok}")
```

You could also use the [DependencyMatcher](https://spacy.io/usage/rule-based-matching#dependencymatcher).

**Reply by AnnemarieWittig** [reply]
Your snipped already did it! Thank you!
